poc_iot_verifier/src/main.rs

```diff
@@ -67,15 +69,61 @@ impl Server {
         });
 
         let gateway_cache = GatewayCache::from_settings(settings).await?;
+
+        let (file_upload_tx, file_upload_rx) = file_upload::message_channel();
+        let file_upload =
+            file_upload::FileUpload::from_settings(&settings.output, file_upload_rx).await?;
+
+        let store_base_path = std::path::Path::new(&settings.cache);
+        // Gateway reward shares sink
+        let (gateway_rewards_tx, gateway_rewards_rx) = file_sink::message_channel(50);
+        let mut gateway_rewards = file_sink::FileSinkBuilder::new(
+            FileType::GatewayRewardShare,
+            store_base_path,
+            gateway_rewards_rx,
+        )
+        .deposits(Some(file_upload_tx.clone()))
+        .write_manifest(true)
```

Comment:
this is a relatively recent addition to the `file_store` app that has been running in the mobile verifier for a little bit now: basically when this method is attached to a file sink and set the `true` the file_sink will keep track of all the files (reward_shares in this case) that are "sunk" so that it can generate the manifest at the end of the reward period (the manifest being a list of all the files that were saved for that period).

the manifest is the indicator to the indexer app downstream that a reward period has "completed"; so you could sink reward files into the downstream bucket all day long but the indexer won't collect them and add rewards to the hotspots' totals until it sees a new manifest file.